### PR TITLE
ASAA-33-Dependency Updates

### PR DESCRIPTION
**Pull request by LuisRocketRangel**
Requirements:
-Update Dependencies to latest versions

Solution:
-Did above

### CODE REVIEW

File: buildSrc/build.gradle.kts [hunk 1]
```diff
@@ -17,8 +17,8 @@ plugins {
 dependencies {
     // This allows usage of kotlin and android gradle plugin apis as well as importing relevant types from the plugins in buildSrc files.
     // Found in https://quickbirdstudios.com/blog/gradle-kotlin-buildsrc-plugin-android/
-    // Unable to use a single source of truth due to issue mentioned in https://github.com/gradle/kotlin-dsl-samples/issues/1320#issuecomment-486309410
+    // Unable to use a   single source of truth due to issue mentioned in https://github.com/gradle/kotlin-dsl-samples/issues/1320#issuecomment-486309410
     // TODO: These should be updated in tandem with source of truth values in Dependencies.kt.
-    implementation("com.android.tools.build:gradle:7.2.0")
-    implementation("org.jetbrains.kotlin:kotlin-gradle-plugin:1.7.0")
+    implementation("com.android.tools.build:gradle:7.2.1")
```
Comment (IvanTelisman) [inline buildSrc/build.gradle.kts]: Hey @LuisRocketRangel, I believe this `com.android.tools.build:gradle` dependency should be set to the same version as the one in Dependencies.kt

File: buildSrc/src/main/kotlin/Dependencies.kt [hunk 2]
```diff
@@ -42,27 +42,27 @@ object Config {
...
+    const val KSP_VERSION = "1.7.20-1.0.8"
 
     /**
      * Called from root project buildscript block in the project root build.gradle.kts
      */
     object BuildScriptPlugins {
         // https://developer.android.com/studio/releases/gradle-plugin
         // TODO: Update corresponding buildSrc/build.gradle.kts value when updating this version!
-        const val ANDROID_GRADLE = "com.android.tools.build:gradle:7.2.0"
+        const val ANDROID_GRADLE = "com.android.tools.build:gradle:7.3.0"
```
Comment (IvanTelisman) [inline buildSrc/src/main/kotlin/Dependencies.kt]: @LuisRocketRangel In reference to the previous comment about this dependency `com.android.tools.build:gradle` in the build.gradle.kts, I believe this is the one that should match up

File: buildSrc/build.gradle.kts [hunk 3]
```diff
@@ -17,8 +17,8 @@ plugins {
 dependencies {
     // This allows usage of kotlin and android gradle plugin apis as well as importing relevant types from the plugins in buildSrc files.
     // Found in https://quickbirdstudios.com/blog/gradle-kotlin-buildsrc-plugin-android/
-    // Unable to use a single source of truth due to issue mentioned in https://github.com/gradle/kotlin-dsl-samples/issues/1320#issuecomment-486309410
+    // Unable to use a   single source of truth due to issue mentioned in https://github.com/gradle/kotlin-dsl-samples/issues/1320#issuecomment-486309410
     // TODO: These should be updated in tandem with source of truth values in Dependencies.kt.
-    implementation("com.android.tools.build:gradle:7.2.0")
-    implementation("org.jetbrains.kotlin:kotlin-gradle-plugin:1.7.0")
```
Comment (jaredstockton) [inline buildSrc/build.gradle.kts:23]: @LuisRocketRangel I'm curious to hear why this dependency was removed :)

File: buildSrc/build.gradle.kts [hunk 4]
```diff
@@ -17,8 +17,8 @@ plugins {
 dependencies {
     // This allows usage of kotlin and android gradle plugin apis as well as importing relevant types from the plugins in buildSrc files.
     // Found in https://quickbirdstudios.com/blog/gradle-kotlin-buildsrc-plugin-android/
-    // Unable to use a single source of truth due to issue mentioned in https://github.com/gradle/kotlin-dsl-samples/issues/1320#issuecomment-486309410
+    // Unable to use a   single source of truth due to issue mentioned in https://github.com/gradle/kotlin-dsl-samples/issues/1320#issuecomment-486309410
```
Comment (jaredstockton) [inline buildSrc/build.gradle.kts]: @LuisRocketRangel nit: Revert this unintentional whitespace change

File: buildSrc/build.gradle.kts (hunk 1 above)
Comment (jaredstockton) [reply buildSrc/build.gradle.kts]: @LuisRocketRangel Agree with what @IvanTelisman mentioned here. Both versions should match

File: buildSrc/src/main/kotlin/Dependencies.kt [hunk 5]
```diff
@@ -10,16 +10,16 @@ import java.util.Locale
...
+private const val KOTLIN_COROUTINES_VERSION = "1.6.4"
 
 /**
  * Provides the source of truth for version/configuration information to any gradle build file (project and app module build.gradle.kts)
  */
 object Config {
     // https://github.com/JLLeitschuh/ktlint-gradle/blob/master/CHANGELOG.md
     // https://github.com/JLLeitschuh/ktlint-gradle/releases
-    const val KTLINT_GRADLE_VERSION = "10.2.1"
+    const val KTLINT_GRADLE_VERSION = "11.0.0"
```
Comment (jaredstockton) [inline buildSrc/src/main/kotlin/Dependencies.kt:22]: @LuisRocketRangel Please run the Android Studio `ktlintFormat` Run Configuration to see if any additional violations are found with the latest version of ktlint

File: buildSrc/src/main/kotlin/Dependencies.kt [hunk 6]
```diff
@@ -42,27 +42,27 @@ object Config {
     //   https://detekt.github.io/detekt/performance.html
     //   https://detekt.github.io/detekt/style.html
     // Release info: https://github.com/detekt/detekt/releases
-    const val DETEKT_VERSION = "1.19.0"
+    const val DETEKT_VERSION = "1.21.0"
```
Comment (jaredstockton) [inline buildSrc/src/main/kotlin/Dependencies.kt]: @LuisRocketRangel Please run the Android Studio `detekt` Run Configuration to see if any additional violations are found with the latest version of detekt

File: buildSrc/src/main/kotlin/Dependencies.kt [hunk 7]
```diff
@@ -96,8 +96,8 @@ object Config {
     }
 
     object Compose {
-        const val COMPOSE_VERSION = "1.2.0-rc03"
-        const val COMPOSE_COMPILER_VERSION = "1.2.0"
+        const val COMPOSE_VERSION = "1.3.0"
```
Comment (jaredstockton) [inline buildSrc/src/main/kotlin/Dependencies.kt:99]: @LuisRocketRangel

1. Have you tested the compose UI manually?
2. Please address any deprecations mentioned in the Compose code (if any)

**Review by jaredstockton — CHANGES_REQUESTED** [review]
@LuisRocketRangel Thanks for working on this! I did leave some comments I'd like to hear back from you on.

Additionally, I just wanted to double check that you've smoke tested the app as well as ran the `ci - internalDebug ...` Android Studio Run Configuration and addresses any issues that came up

File: buildSrc/build.gradle.kts (hunk 3 above)
Comment (LuisRocketRangel) [reply buildSrc/build.gradle.kts:23]: That was unintentional I will fix this

**Comment by LuisRocketRangel** [comment]
Hello, I did run KtlintFormat -> KtlintCheck -> internalDebug like I usually do and it seemed like things were okay. I will address all the feedback

File: buildSrc/src/main/kotlin/Dependencies.kt (hunk 7 above)
Comment (LuisRocketRangel) [reply buildSrc/src/main/kotlin/Dependencies.kt:99]: Hey I re upped the PR, check it out!

**Review by IvanTelisman — APPROVED** [review]

**Review by jaredstockton — APPROVED** [review]
@LuisRocketRangel I just pushed a review commit with a few additional changes primarily around detekt. Appreciate the help on this!

**Pull request merged**